feat: Query vm hostname

handsomexdd1024 opened the pull request:
Add support for querying vm hostname via qemu-ga.

API Path at /nodes/{node}/qemu/{vmid}/agent/get-host-name

handsomexdd1024 committed:
feat: Query vm hostname

handsomexdd1024 commented on virtual_machine.go:
```diff
@@ -467,6 +467,26 @@ func (v *VirtualMachine) MoveDisk(ctx context.Context, disk string, params *Virt
 	return NewTask(upid, v.client), nil
 }
 
+func (v *VirtualMachine) AgentGetHostName(ctx context.Context) (hostname *string, err error) {
```
Yeah you're right. I'll fix this right away.

handsomexdd1024 replied:
No wait, what if an error occured and operation failed? Just return an empty string instead of `nil`?

luthermonson replied:
yes of course, what's wrong with an empty string and an err? still a string... and you err check before you do anything dont you?

luthermonson replied:
let me elaborate, if you use a pointer to a string you are creating a ternary, it's nil, empty string or a string with value... if you err do you return nil? do you return an empty string? or do you return "error"? does returning nil mean something more significant than empty string? you're adding another paradigm that's probably not necessary for a func this simple

luthermonson committed:
AgentGetHostName: return string instead of *string

Per PR review: a pointer-to-string return creates a tristate
(nil/empty/value) that's unnecessary here. Empty string + err is
sufficient since callers check err first.
